### PR TITLE
APM-4586 update proxygen spec

### PR DESCRIPTION
**Pull request by harryhmc**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/proxygen.yaml [hunk 1]
```diff
@@ -1219,6 +1221,14 @@ paths:
         name: environment
         in: path
         required: true
+      - in: path
```
Comment (acarriedev) [inline specification/proxygen.yaml:1224]: I'm not sure this is valid as we're explicitly calling out apikey and later mtls then this secret_type doesn't refer to anything in the path

File: specification/proxygen.yaml [hunk 2]
```diff
@@ -1219,6 +1221,14 @@ paths:
...
+        name: secret_type
+        - schema:
+            type: string
+            enum:
+              - apikey
+              - mtls
+        required: true
       - in: path
         name: secret_name
```
Comment (acarriedev) [inline specification/proxygen.yaml:1233]: I think we need to call out that this is a string here

File: specification/proxygen.yaml [hunk 3]
```diff
@@ -1260,6 +1270,98 @@ paths:
                 empty-request-body:
                   value:
                     detail: Request body cannot be empty
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPException'
+              examples:
```
Comment (acarriedev) [inline specification/proxygen.yaml:1279]: These examples aren't valid for API key (I'm not sure what a bad request would come up as for api key - might not need it here?)

File: specification/proxygen.yaml [hunk 4]
```diff
@@ -1260,6 +1270,98 @@ paths:
...
+                  value:
+                    detail: Invalid key file in payload. key file cannot be password protected.
+      requestBody:
+        required: true
+        content:
+          text/plain:
+            schema:
+              type: string
+              minLength: 1
+            examples: {}
```
Comment (acarriedev) [inline specification/proxygen.yaml:1293]: worth a simple example?

File: specification/proxygen.yaml [hunk 5]
```diff
@@ -1260,6 +1270,98 @@ paths:
...
+        name: api_name
+        in: path
+        required: true
+      - schema:
+          type: string
+        name: environment
+        in: path
+        required: true
+      - in: path
+        name: secret_type
```
Comment (acarriedev) [inline specification/proxygen.yaml:1317]: See previous comment about this

File: specification/proxygen.yaml [hunk 6]
```diff
@@ -1300,29 +1397,48 @@ paths:
...
+        Adding this secret creates or updates a keystore on the API platform which is used in a handshake with the target server certificate, which verifies that both the client and server certificates have been signed by the same root certificate authority.
 
-        > * When setting the query parameter `type=mtls` the certificate and private key files should be in PEM format and must be sent as `multipart/form-data` under fields of `key` and `cert`.
+        > * The certificate and private key files should be in PEM format and must be sent as `multipart/form-data` under fields of `key` and `cert`.
         > * Updating an mtls secret will always update the keystore and will be used by any instances of the API in the specified environment. This is can be used for key rotation but should be done with caution for live environments.
+  '/apis/{api_name}/environments/{environment}/secrets/{secret_type}/{secret_name}':
```
Comment (acarriedev) [inline specification/proxygen.yaml:1413]: Is this needed? It may be but we're calling out both valid options above

**Review by acarriedev — CHANGES_REQUESTED** [review]

File: specification/proxygen.yaml (hunk 2 above)
Comment (harryhmc) [reply specification/proxygen.yaml:1233]: we do! check the `schema` on the following line

File: specification/proxygen.yaml (hunk 6 above)
Comment (harryhmc) [reply specification/proxygen.yaml:1413]: this is here for the get / delete methods which share request/response syntax regardless of the secret type

Comment (acarriedev) [reply specification/proxygen.yaml:1384]: Got you - makes sense ta

**Review by acarriedev — APPROVED** [review]

**Review by acarriedev — APPROVED** [review]

**Pull request merged**